tests/scope_test.js

```diff
@@ -0,0 +1,577 @@
+require('./helper');
+
+var expect = require('chai').expect;
+var Sequelize = require('..');
+var DataTypes = Sequelize.DataTypes;
+var Op = Sequelize.Op;
+
+describe('associations', () => {
+  describe('scope', () => {
+    beforeEach(function() {
+      this.Post = this.sequelize.define('post', {});
+      this.Image = this.sequelize.define('image', {});
+      this.Question = this.sequelize.define('question', {});
+      this.Comment = this.sequelize.define('comment', {
+        title: Sequelize.STRING,
+        type: Sequelize.STRING,
+        commentable: Sequelize.STRING,
+        commentable_id: Sequelize.INTEGER,
+        isMain: {
+          type: Sequelize.BOOLEAN,
+          defaultValue: false
+        }
+      });
+
+      this.Comment.prototype.getItem = function() {
+        return this[`get${this.get('commentable').substr(0, 1).toUpperCase()}${this.get('commentable').substr(1)}`]();
+      };
+
+      this.Post.addScope('withComments', {
+        include: [this.Comment]
+      });
+      this.Post.addScope('withMainComment', {
+        include: [{
+          model: this.Comment,
+          as: 'mainComment'
+        }]
+      });
+      this.Post.hasMany(this.Comment, {
+        foreignKey: 'commentable_id',
+        scope: {
+          commentable: 'post'
+        },
+        constraints: false
+      });
+      this.Post.hasMany(this.Comment, {
+        foreignKey: 'commentable_id',
+        as: 'coloredComments',
+        scope: {
+          commentable: 'post',
+          type: { [Op.in]: ['blue', 'green'] }
+        },
+        constraints: false
+      });
+      this.Post.hasOne(this.Comment, {
+        foreignKey: 'commentable_id',
+        as: 'mainComment',
+        scope: {
+          commentable: 'post',
+          isMain: true
+        },
+        constraints: false
+      });
+      this.Comment.belongsTo(this.Post, {
+        foreignKey: 'commentable_id',
+        as: 'post',
+        constraints: false
+      });
+
+      this.Image.hasMany(this.Comment, {
+        foreignKey: 'commentable_id',
+        scope: {
+          commentable: 'image'
+        },
+        constraints: false
+      });
+      this.Comment.belongsTo(this.Image, {
+        foreignKey: 'commentable_id',
+        as: 'image',
+        constraints: false
+      });
+
+      this.Question.hasMany(this.Comment, {
+        foreignKey: 'commentable_id',
+        scope: {
+          commentable: 'question'
+        },
+        constraints: false
+      });
+      this.Comment.belongsTo(this.Question, {
+        foreignKey: 'commentable_id',
+        as: 'question',
+        constraints: false
+      });
+    });
+
+    // Reason: Fixed in another PR: https://github.com/cockroachdb/sequelize-cockroachdb/pull/64
```

Comment:
Are we able to unskip this if it was fixed?

Reply:
Probably. I'll check it out.